src/firebase/firebase-sn.ts

```diff
@@ -1,28 +1,115 @@
 import {
   byKey, firstByChildKey, byChildKey
 } from './collections'
+import {append, reduce} from 'ramda'
+import {Test} from "tape-async";
+import {test} from "../test/test";
 
 interface FirebaseValue {
-  [propName:string]:FirebaseValue | string | boolean | number
+  [propName:string]:FirebaseValue | string | boolean | number | FirebaseValue
 }
 
-export function Model(model:string) {
-  return function(seneca) {
-    return {
-      push: function(values:FirebaseValue):Promise<KeyResponse> {
-         return seneca.act('role:Firebase,cmd:push', {model, values})
-      },
-      update: function(key:string, values:FirebaseValue):Promise<KeyResponse> {
-        return seneca.act('role:Firebase,cmd:update', {model, key, values})
-      },
-      remove: function(key:string):Promise<KeyResponse> {
-        return seneca.act('role:Firebase,cmd:remove', {model, key})
-      }
+class Updater {
```

Comment:
Looking at this code I'm definitely feeling that this whole file should go away at some point, or at least no longer be a seneca service.